Tests/SimpleHTTPTests/Encoder/MultipartFormDataEncoderTests.swift

```diff
@@ -0,0 +1,158 @@
+import XCTest
+@testable import SimpleHTTP
+
+class MultipartFormDataEncoderTests: XCTestCase {
+
+    let crlf = EncodingCharacters.crlf
+
+    func test_encode_multipartAddData_bodyPart() throws {
+        // Given
+        let boundary = "boundary"
+        var multipart = MultipartFormData(boundary: boundary)
+
+        let data = "I'm pjechris, Nice to meet you"
+        let name = "data"
+        multipart.add(data: Data(data.utf8), name: name)
+
+        let expectedString = (
+            Boundary.string(for: .initial, boundary: boundary)
+            + "Content-Disposition: form-data; name=\"\(name)\"\(crlf)\(crlf)"
+            + data
+            + Boundary.string(for: .final, boundary: boundary)
+        )
+        let expectedData = Data(expectedString.utf8)
+
+        // When
+        let encodedData = try MultipartFormDataEncoder(body: multipart).encode()
+
+        // Then
+        XCTAssertEqual(encodedData, expectedData)
+    }
+
+
+    func test_encoding_data_multipleBodyPart() throws {
+        let boundary = "boundary"
+        var multipart = MultipartFormData(boundary: boundary)
+
+        let data1 = "Swift"
+        let name1 = "swift"
+        multipart.add(data: Data(data1.utf8), name: name1)
+
+        let data2 = "Combine"
```

Comment:
maybe you could levereage tuples, WDYT?

`let part2 = (data: "Combine", name: "combine", mimeType: "")`

Reply:
`name` and `data` are use in the `expectedString`. And the name could be change. I think it's better to separate them